R/board_rsconnect.R

```diff
@@ -549,20 +550,33 @@ rsc_user_name <- function(board, guid) {
 }
 
 read_cache <- function(path) {
-  if (file.exists(path)) {
+  if (file.exists(path) && use_cache()) {
     yaml::read_yaml(path, eval.expr = FALSE)
   } else {
     list()
   }
 }
-update_cache <- function(path, key, value) {
-  cache <- read_cache(path)
-  cache[[key]] <- value
-  write_yaml(cache, path)
 
+update_cache <- function(path, key, value) {
+  if (use_cache()) {
```

Comment:
I am checking the cache option in _both_ `read_cache()` and `update_cache()` even though the second calls the first so we avoid writing to the cache here in a couple of lines ⬇️